lib/Core/Executor.cpp

```diff
@@ -3706,14 +3706,21 @@ void Executor::executeMemoryOperation(ExecutionState &state, bool isWrite,
     ref<Expr> offset = mo->getOffsetExpr(address);
     ref<Expr> boundsCheck = mo->getBoundsCheckOffset(offset, bytes);
 
+
     bool inBounds;
-    solver->setTimeout(coreSolverTimeout);
-    bool success = solver->mustBeTrue(state, boundsCheck, inBounds);
-    solver->setTimeout(0);
-    if (!success) {
-      state.pc = state.prevPC;
-      terminateStateEarly(state, "Query timed out (bounds check).");
-      return;
+    if(NoBoundCheck){
+    	// No Bounds check is needed to be performed. So, inBounds is set to
+    	// be true. This is used when the user wants to get the skeleton tree.
+    	inBounds = true;
+    }else{
+    	solver->setTimeout(coreSolverTimeout);
```

Comment:
This section is not properly indented: Tip: can use `git clang-format` to automatically properly format the code.